From slony1-bugs at lists.slony.info  Tue Sep  9 14:26:45 2008
From: slony1-bugs at lists.slony.info (slony1-bugs@lists.slony.info)
Date: Tue Sep  9 14:26:47 2008
Subject: [Slony1-bugs] [Bug 56] New: store trigger runs against all nodes
	upon subscription
Message-ID: <bug-56-4@http.www.slony.info/bugzilla/>

http://www.slony.info/bugzilla/show_bug.cgi?id=56

           Summary: store trigger runs against all nodes upon subscription
           Product: Slony-I
           Version: 1.2
          Platform: PC
        OS/Version: Linux
            Status: ASSIGNED
          Severity: major
          Priority: high
         Component: slon
        AssignedTo: slony1-bugs@lists.slony.info
        ReportedBy: cbbrowne@ca.afilias.info
                CC: slony1-bugs@lists.slony.info
   Estimated Hours: 0.0


As part of COPY SET, the new subscriber node is supposed to copy the contents
of sl_trigger from its provider, and run StoreTrigger() against each entry so
that any triggers that are to run on subscribers are configured to run.

Unfortunately, what is run is StoreTrigger(), rather than StoreTrigger_int().

StoreTrigger_int() runs only against the local node, which is what we want.

StoreTrigger() starts by generating a STORE TRIGGER event, so that this action
winds up being run against all nodes.  This means that, as part of COPY_SET,
you've got a table alteration running against all nodes.  This has the
potential to lead to Unexpected Locking of a provider node.


-- 
Configure bugmail: http://www.slony.info/bugzilla/userprefs.cgi?tab=email
------- You are receiving this mail because: -------
You are on the CC list for the bug.
You are the assignee for the bug.
From slony1-bugs at lists.slony.info  Tue Sep  9 14:27:31 2008
From: slony1-bugs at lists.slony.info (slony1-bugs@lists.slony.info)
Date: Tue Sep  9 14:27:32 2008
Subject: [Slony1-bugs] [Bug 56] store trigger runs against all nodes upon
	subscription
In-Reply-To: <bug-56-4@http.www.slony.info/bugzilla/>
Message-ID: <20080909212731.4E491290EB7@main.slony.info>

http://www.slony.info/bugzilla/show_bug.cgi?id=56





--- Comment #1 from Christopher Browne <cbbrowne@ca.afilias.info>  2008-09-09 14:27:31 ---
Created an attachment (id=21)
 --> (http://www.slony.info/bugzilla/attachment.cgi?id=21)
Patch for 1.2 branc


-- 
Configure bugmail: http://www.slony.info/bugzilla/userprefs.cgi?tab=email
------- You are receiving this mail because: -------
You are on the CC list for the bug.
You are the assignee for the bug.
From slony1-bugs at lists.slony.info  Tue Sep  9 14:28:13 2008
From: slony1-bugs at lists.slony.info (slony1-bugs@lists.slony.info)
Date: Tue Sep  9 14:28:14 2008
Subject: [Slony1-bugs] [Bug 56] store trigger runs against all nodes upon
	subscription
In-Reply-To: <bug-56-4@http.www.slony.info/bugzilla/>
Message-ID: <20080909212813.EF548290EBD@main.slony.info>

http://www.slony.info/bugzilla/show_bug.cgi?id=56





--- Comment #2 from Christopher Browne <cbbrowne@ca.afilias.info>  2008-09-09 14:28:14 ---
Note that as the handling of triggers changes fundamentally in v2.0, this is
not an issue in that version of Slony-I.


-- 
Configure bugmail: http://www.slony.info/bugzilla/userprefs.cgi?tab=email
------- You are receiving this mail because: -------
You are on the CC list for the bug.
You are the assignee for the bug.
From slony1-bugs at lists.slony.info  Wed Sep 24 11:57:01 2008
From: slony1-bugs at lists.slony.info (slony1-bugs@lists.slony.info)
Date: Wed Sep 24 11:57:03 2008
Subject: [Slony1-bugs] [Bug 46] New 8.4 #include - utils/snapmgr.h
In-Reply-To: <bug-46-4@http.www.slony.info/bugzilla/>
Message-ID: <20080924185701.9F932290F4D@main.slony.info>

http://www.slony.info/bugzilla/show_bug.cgi?id=46





--- Comment #3 from Christopher Browne <cbbrowne@ca.afilias.info>  2008-09-24 11:57:00 ---
Created an attachment (id=22)
 --> (http://www.slony.info/bugzilla/attachment.cgi?id=22)
Newer patch to Slony-I to run against 8.4 - uses new GetActiveSnapshot()
function

Attached is a newer patch.

This needs to be modified somewhat to address the fact that 8.3 expects the
former form, and 8.4 expects the new form.  I'll discuss that on-list.


-- 
Configure bugmail: http://www.slony.info/bugzilla/userprefs.cgi?tab=email
------- You are receiving this mail because: -------
You are on the CC list for the bug.
You are the assignee for the bug.
From slony1-bugs at lists.slony.info  Wed Sep 24 12:56:32 2008
From: slony1-bugs at lists.slony.info (slony1-bugs@lists.slony.info)
Date: Wed Sep 24 12:56:34 2008
Subject: [Slony1-bugs] [Bug 46] New 8.4 #include - utils/snapmgr.h
In-Reply-To: <bug-46-4@http.www.slony.info/bugzilla/>
Message-ID: <20080924195632.91511290F3D@main.slony.info>

http://www.slony.info/bugzilla/show_bug.cgi?id=46


Christopher Browne <cbbrowne@ca.afilias.info> changed:

           What    |Removed                     |Added
----------------------------------------------------------------------------
         Resolution|                            |FIXED
             Status|NEW                         |RESOLVED




--- Comment #4 from Christopher Browne <cbbrowne@ca.afilias.info>  2008-09-24 12:56:32 ---
Resolved in the CVS commit that involved the following...

http://lists.slony.info/pipermail/slony1-commit/2008-September/002366.html

- Autoconf detects whether or not the new function is available
- If it is, config.h.in defines a preprocessor variable
- If that variable is defined, slony1_funcs.c uses the new function


-- 
Configure bugmail: http://www.slony.info/bugzilla/userprefs.cgi?tab=email
------- You are receiving this mail because: -------
You are on the CC list for the bug.
You are the assignee for the bug.
From cbbrowne at ca.afilias.info  Thu Sep 25 08:32:29 2008
From: cbbrowne at ca.afilias.info (Christopher Browne)
Date: Thu Sep 25 08:32:37 2008
Subject: [Slony1-bugs] Cache lookup failure when starting slony
In-Reply-To: <47FB6578.9060006@streamtech.nl> (Gersom van de Bunt's message of
	"Tue, 08 Apr 2008 14:30:48 +0200")
References: <47FB6578.9060006@streamtech.nl>
Message-ID: <871vz8qm0i.fsf@dba2.int.libertyrms.com>

Gersom van de Bunt <gersom@streamtech.nl> writes:
> cache lookup failed for type 76787

That sounds a lot like the following issue, which was reasonably
well-discussed:

http://lists.slony.info/pipermail/slony1-general/2008-June/008237.html
-- 
"cbbrowne","@","ca.afilias.info"
<http://dba2.int.libertyrms.com/>
Christopher Browne
(416) 673-4124 (land)
From slony1-bugs at lists.slony.info  Thu Sep 25 09:28:05 2008
From: slony1-bugs at lists.slony.info (slony1-bugs@lists.slony.info)
Date: Thu Sep 25 09:28:06 2008
Subject: [Slony1-bugs] [Bug 57] New: Missing data and syncs ignored
Message-ID: <bug-57-4@http.www.slony.info/bugzilla/>

http://www.slony.info/bugzilla/show_bug.cgi?id=57

           Summary: Missing data and syncs ignored
           Product: Slony-I
           Version: 1.2
          Platform: Other
        OS/Version: Linux
            Status: NEW
          Severity: major
          Priority: medium
         Component: slon
        AssignedTo: slony1-bugs@lists.slony.info
        ReportedBy: meetesh.karia@alumni.duke.edu
                CC: slony1-bugs@lists.slony.info
   Estimated Hours: 0.0


We've been running slony (currently on version 1.2.14) for a while and recently
we ran into a situation where a SYNC was failing due to a constraint violation
exception.  After looking into why that happened, I was able to see that some
deletes that occurred on the master were never propagated to the slave and that
then caused the constraint violation further down the road.

To add to the mysteriousness, after the SYNC had been tried numerous times, it
finally just decided to do nothing for the SYNC and continue on.

Our setup has a postgres 8.3.1 db running on the master with a slave (slave 1)
on the same LAN.  We also have a slave (slave 2) in another environment with
that replication happening over a WAN (there is a separate slon daemon running
in the other environment).  Events were going from the master to slave 1 and
then slave 1 was set up as a provider for slave 2.

This may or may not be an issue, but we've occasionally seen network issues
between our two environments.  However, the issue occurred with slave 1 so it
happened before it got to the network between the two environments.

Here is a snippet of the log file showing how the SYNC was ignored:

2008-09-24 23:04:32 CDT DEBUG2 remoteWorkerThread_3: SYNC 262954 processing
2008-09-24 23:04:32 CDT DEBUG2 remoteWorkerThread_3: syncing set 3 with 3
table(s) from provider 3
2008-09-24 23:04:32 CDT DEBUG2  ssy_action_list length: 0
2008-09-24 23:04:32 CDT DEBUG2 remoteWorkerThread_3: syncing set 5 with 1
table(s) from provider 3
2008-09-24 23:04:32 CDT DEBUG2  ssy_action_list length: 0
2008-09-24 23:04:32 CDT DEBUG2 remoteWorkerThread_3: syncing set 2 with 1
table(s) from provider 3
2008-09-24 23:04:32 CDT DEBUG2  ssy_action_list length: 0
2008-09-24 23:04:32 CDT DEBUG2 remoteWorkerThread_3: syncing set 1 with 51
table(s) from provider 3
2008-09-24 23:04:32 CDT DEBUG2  ssy_action_list length: 0
2008-09-24 23:04:32 CDT DEBUG2 remoteWorkerThread_3: current local log_status
is 0
2008-09-24 23:04:32 CDT DEBUG2 remoteWorkerThread_3_3: current remote
log_status = 0
2008-09-24 23:04:32 CDT DEBUG2 remoteHelperThread_3_3: 0.056 seconds delay for
first row
2008-09-24 23:04:32 CDT DEBUG2 remoteHelperThread_3_3: 0.063 seconds until
close cursor
2008-09-24 23:04:32 CDT DEBUG2 remoteHelperThread_3_3: inserts=49 updates=0
deletes=0
2008-09-24 23:04:32 CDT ERROR  remoteWorkerThread_3: <snip>
" ERROR:  duplicate key value violates unique constraint
"idx_unique_calendar_entry"
 - qualification was:
2008-09-24 23:04:32 CDT ERROR  remoteWorkerThread_3: SYNC aborted
2008-09-24 23:04:33 CDT DEBUG2 remoteListenThread_3: queue event 3,263077 SYNC
2008-09-24 23:04:33 CDT DEBUG2 remoteListenThread_3: UNLISTEN
2008-09-24 23:04:35 CDT DEBUG2 remoteListenThread_4: queue event 4,107043 SYNC
2008-09-24 23:04:35 CDT DEBUG2 remoteWorkerThread_4: Received event 4,107043
SYNC
2008-09-24 23:04:35 CDT DEBUG2 calc sync size - last time: 1 last length: 10053
ideal: 5 proposed size: 3
2008-09-24 23:04:35 CDT DEBUG2 remoteWorkerThread_4: SYNC 107043 processing
2008-09-24 23:04:35 CDT DEBUG2 remoteWorkerThread_4: no sets need syncing for
this event
2008-09-24 23:04:38 CDT DEBUG2 syncThread: new sl_action_seq 1 - SYNC 107646
2008-09-24 23:04:39 CDT DEBUG2 localListenThread: Received event 1,107646 SYNC
2008-09-24 23:04:40 CDT DEBUG2 remoteWorkerThread_4: forward confirm 4,107043
received by 3
2008-09-24 23:04:41 CDT DEBUG2 remoteListenThread_3: LISTEN
2008-09-24 23:04:42 CDT DEBUG2 remoteWorkerThread_3: SYNC 262954 processing
2008-09-24 23:04:42 CDT DEBUG2 remoteWorkerThread_3: syncing set 3 with 3
table(s) from provider 3
2008-09-24 23:04:42 CDT DEBUG2  ssy_action_list length: 0
2008-09-24 23:04:42 CDT DEBUG2 remoteWorkerThread_3: syncing set 5 with 1
table(s) from provider 3
2008-09-24 23:04:42 CDT DEBUG2  ssy_action_list length: 0
2008-09-24 23:04:42 CDT DEBUG2 remoteWorkerThread_3: syncing set 2 with 1
table(s) from provider 3
2008-09-24 23:04:42 CDT DEBUG2  ssy_action_list length: 0
2008-09-24 23:04:42 CDT DEBUG2 remoteWorkerThread_3: syncing set 1 with 51
table(s) from provider 3
2008-09-24 23:04:42 CDT DEBUG2  ssy_action_list length: 0
2008-09-24 23:04:42 CDT DEBUG2 remoteWorkerThread_3: current local log_status
is 0
2008-09-24 23:04:42 CDT DEBUG2 remoteWorkerThread_3_3: current remote
log_status = 0
2008-09-24 23:04:42 CDT DEBUG2 remoteHelperThread_3_3: 0.018 seconds delay for
first row
2008-09-24 23:04:42 CDT DEBUG2 remoteHelperThread_3_3: 0.034 seconds until
close cursor
2008-09-24 23:04:42 CDT DEBUG2 remoteHelperThread_3_3: inserts=0 updates=0
deletes=0
2008-09-24 23:04:42 CDT DEBUG2 remoteWorkerThread_3: SYNC 262954 done in 0.055
seconds


-- 
Configure bugmail: http://www.slony.info/bugzilla/userprefs.cgi?tab=email
------- You are receiving this mail because: -------
You are on the CC list for the bug.
You are the assignee for the bug.
